### ISSUE TITLE
Proxy settings from config.json ignored

### ISSUE THREAD

**Issue by elvinasp** [bmuschko/gradle-docker-plugin#540]
Hello, 

I am trying to build custom Docker images for my project and image build process fails if run in an environment under proxy. I have already resolved proxy issues with Docker and Gradle itself by providing PROXY configuration to Docker Service and CLI arguments to Gradle. Now there is a network connectivity issue from Docker image itself during build process. 

I have set proxy configuration in ~/.docker/config.json like this:
`{
  "auths":   { },
  "HttpHeaders": {"User-Agent": "Docker-Client/17.09.0-ce (linux)"},
  "proxies": {"default":   {
    "httpProxy": "http://CORP_PROXY:3128",
    "httpsProxy": "http://CORP_PROXY:3128",
    "noProxy": "127.0.0.1,localhost"
  }}
}`

**Note** Docker plugin requires to have auths section even if no auth is used, otherwise plugin fails with invalid Auth error. 

PROXY environment is set properly if image is started directly via Docker. It does not work if image is started via Gradle plugin. 

**QUESTION**:  Is this functionality not implemented or do I have to specify those parameters explicitly in build.gradle? I would like to avoid modifying build.gradle and just configure build environment appropriately instead. 

Build process stalls for a long time and after that I do see a network timeout 
`09:00:34.157 [QUIET] [org.gradle.api.Task] Step 1/12 : FROM postgres:9.4
09:00:50.167 [QUIET] [org.gradle.api.Task] ---> e0c7250b6ea3
09:00:50.168 [QUIET] [org.gradle.api.Task] Step 2/12 : MAINTAINER devops@COMPANY.COM
09:00:50.906 [QUIET] [org.gradle.api.Task] ---> Running in b046337436fe
09:00:53.466 [QUIET] [org.gradle.api.Task] ---> 66809413ddc0
09:00:53.485 [QUIET] [org.gradle.api.Task] Removing intermediate container b046337436fe
09:00:53.485 [QUIET] [org.gradle.api.Task] Step 3/12 : RUN apt-get update && apt-get install -y less net-tools
09:00:54.018 [QUIET] [org.gradle.api.Task] ---> Running in **312c26b5e7ea**
09:08:55.177 [QUIET] [org.gradle.api.Task] Err http://deb.debian.org jessie InRelease
  
Err http://deb.debian.org jessie-updates InRelease
  
Err http://apt.postgresql.org jessie-pgdg InRelease
09:08:55.178 [QUIET] [org.gradle.api.Task] Err http://deb.debian.org jessie Release.gpg
  Unable to connect to deb.debian.org:http: [IP: 149.20.4.15 80]
Err http://apt.postgresql.org jessie-pgdg Release.gpg
  Unable to connect to apt.postgresql.org:http: [IP: 204.145.124.244 80]
Err http://deb.debian.org jessie-updates Release.gpg
  Unable to connect to deb.debian.org:http: [IP: 149.20.4.15 80]
09:08:55.557 [QUIET] [org.gradle.api.Task] Err http://security.debian.org jessie/updates InRelease
... SKIPPED SOME Failed to fetch / failed to download ERRORS ...
W: Failed to fetch http://apt.postgresql.org/pub/repos/apt/dists/jessie-pgdg/Release.gpg  Unable to connect to apt.postgresql.org:http: [IP: 204.145.124.244 80]
W: Failed to fetch http://security.debian.org/dists/jessie/updates/Release.gpg  Unable to connect to security.debian.org:http: [IP: 149.20.4.14 80]
W: Some index files failed to download. They have been ignored, or old ones used instead.
`

Manually started image, used for intermediary build container has proper PROXY environment configuration injected into it. 
[direct-image-start.txt](https://github.com/bmuschko/gradle-docker-plugin/files/1739572/direct-image-start.txt)

Intermediary container created during build does not have PROXY settings defined. 

[intermediary-container.txt](https://github.com/bmuschko/gradle-docker-plugin/files/1739573/intermediary-container.txt)

**Comment by cdancy** [comment]
@elvinasp take a look at [THIS](https://github.com/bmuschko/gradle-docker-plugin/issues/462) issue and see if it helps you out any.

**Comment by elvinasp** [comment]
Yes, I have searched issue list before posting and saw that link. I have had the same problem with image download and solved it by passing -Dhttp* options. 

However this is a bit different issue. Same PROXY settings have to be injected into docker container environment started from downloaded image in order for the container to access internet. This can be done manually, by altering Dockerfile and as a result build.gradle or this can be done automatically by Docker system itself via ~/.docker/config.json. I have checked that by manually starting a container from image from cli. But intermediary container started by Docker plugin does not get proxy details injected into environment. 

Updating build.gradle would increase its complexity as build have to work in both proxied and non proxied environment. And as code freeze is scheduled at end of February I would like to avoid changing build.gradle. :)

FIY: [Palantir plugin](https://github.com/palantir/gradle-docker) works properly with config.json. My colleague is using Palantir plugin in his project and he suggested to add proxy info to config.json, but as I mentioned previously I would like to avoid messing with build.gradle at the moment.
